specification/devcenter/DevCenter/Environments/models.tsp

```diff
@@ -118,23 +185,130 @@ model Environment {
 
   @doc("Provisioning error details. Populated only for error states.")
   @visibility("read")
-  error?: Azure.Core.Foundations.Error;
+  @typeChangedFrom(APIVersions.v2024_02_01, Azure.Core.Foundations.Error)
+  error?: CloudErrorBody;
 }
 
 @doc("""
 Properties of an environment. These properties can be updated after the
 resource has been created.
 """)
-model EnvironmentUpdateProperties {
+model EnvironmentUpdateProperties is EnvironmentPatchProperties {
```

Comment:
ok, I was a little confused here, so I went to look the docs and swaggers. `is` here is just adding the properties from `EnvironmentPatchProperties` to  `EnvironmentUpdateProperties`, look at [this](https://typespec.io/docs/language-basics/models#property-ordering) examples.  But since `EnvironmentPatchProperties` is not re-used anywhere i found it weird having to create a separate model for that. 

Looking at the [previous swagger](https://github.com/Azure/azure-rest-api-specs/blob/6a3f3be265ebb431fb4b26f318c749b0252bc91e/specification/devcenter/data-plane/Microsoft.DevCenter/stable/2024-02-01/environments.json#L1084) they were added as a `allOf`, not as properties. Then I found [this doc](https://typespec.io/docs/getting-started/typespec-for-openapi-dev#allof-and-polymorphism-using-allof). I think we should try `extends` here, so it's like `EnvironmentUpdateProperties extends EnvironmentPatchProperties`. Let me know if it works that way

Reply:
I read more about this and I believe the previous swagger is using `allOf` but there is no reason for doing it so. To best describe our service:
since PatchProperties can also be updated with PUT we should have
``` 
EnvironmentUpdateProperties {
      ...EnvironmentPatchProperties
}
```
You should be able to add the @added(version) to the patch properties this way too

Reply:
we still need to know what decorator to add to each of those properties considering the put/patch behavior during update

Reply:
> I read more about this and I believe the previous swagger is using `allOf` but there is no reason for doing it so. To best describe our service: since PatchProperties can also be updated with PUT we should have
> 
> ```
> EnvironmentUpdateProperties {
>       ...EnvironmentPatchProperties
> }
> ```
> 
> You should be able to add the @added(version) to the patch properties this way too

Looks like this works, thanks!